src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibility.java

```diff
@@ -551,6 +550,10 @@ public void run() {
                 if (pac == null) return;
                 AccessibleSelection as = pac.getAccessibleSelection();
                 if (as == null) return;
+                if (parent instanceof JList) {
+                    ((JList) parent).setSelectedIndex(i);
+                    return;
+                }
```

Comment:
Looks like the a11y interface miss "setSelectedIndex" method? The code above will not work for the custom component which uses the "AccessibleJListChild" a11y object?

Reply:
Unfortunately, the current a11y interface does not allow this action.
As for the custom AccessibleJListChild, I prepared a [sample project](https://github.com/savoptik/AccessibilityJListItemA11yExample) in which I implemented a custom child. The experiment showed that there are no negative consequences. The custom child works as intended.

Reply:
In the example, you use a JList class, and the checks in the code above works, but if some other "AccessibleJList" class will be used then it will not work, right? This is the reason why implementations of a11y classes usually use specific interfaces like "AccessibleJList".

Reply:
Yes, this will not work with a custom list not inherited from JList. Unfortunately AccessibleJList is not perfect and the functionality you need is not there.
If you have a better solution, please share it. - Could you please advice on what we can do here?

Reply:
This is what I meant in the first message, "a11y interface miss setSelectedIndex method". I suggest filing a separate issue to accumulate such use cases and then fix them by adding additional methods to a11y interfaces. Probably something similar will be found in the Table/etc.

Reply:
Done. [JDK-8271846](https://bugs.openjdk.java.net/browse/JDK-8271846)